### ISSUE TITLE
Inquiry regarding Baseline Implementations (AttnLRP / GradCAM)

### ISSUE THREAD

**Issue by joelromanky** [xmed-lab/TAM#17]
Hi authors,

First, thank you for the great work on TAM. I am currently working on a project involving XAI for VLLMs and I am looking to compare TAM results against the baselines you used.

I noticed in the Appendix that you utilized official repositories for **AttnLRP** and **GradCAM**. Since those frameworks are generally designed for unimodal vision models (like ViT or ResNet), I ran into some difficulties adapting them to the VLLM architecture (QwenVL) especially how the attribution should be computed (wrt to `pixel_values` or the visual tokens after the projector)

**I have two quick questions:**
1. Is the source code for these specific baseline adaptations available, or could it be added to this repository?
2. If sharing the code isn't possible, could you provide some intuition on how you it ?

Any snippets or high-level guidance would be greatly appreciated!

Best regards,

**Comment by Eli-YiLi** [comment]
Thanks for your interest.

I use the core code of AttnLRP and GradCAM from the official repo, and adjust them in my experimental code.
Specifically, I record the attention weights,  logits and apply their code to generate the results.
In my experimental code, I record the attention weights and logits by modifying their modeling file.
So, if you want to implement it, you are suggested to record this hidden weights/features.

1. The experimental code is not refined, not easy to read, so I will not release it.
2. Please refer to the above comments for your implementation. (all the baselines can implement like this)
